### PR TITLE
Adding new terminology

### PR DESCRIPTION
**Pull request by topirissanen-atoz**
Added support for creating new terminologies in this PR. Creating a new terminology can be accessed from front page for logged in users. 

If user is logged in, a button to add new terminologies is shown in front page:
image

User is prompted to choose input type between manual or file upload (manual input added only in this PR):
image

Part of inputs in the form:
image

Required information that is missing is informed to user if they try to create new terminology:
image

### CODE REVIEW

File: src/common/components/new-terminology/index.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  useEffect(() => {
+    if (newVocabulary.isSuccess) {
+      setShowModal(false);
+      dispatch(terminologySearchApi.util.invalidateTags(['TerminologySearch']));
+    }
+  }, [newVocabulary, dispatch]);
+
+  if (!user.superuser) {
+    return null;
+  }
```
Comment (kkyttala) [inline src/common/components/new-terminology/index.tsx:49]: What is the purpose of this check? Users with other roles, like terminology_editor are also allowed to create new terminologies

File: src/common/components/new-terminology/info-file.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      onDragOver={(e) => e.preventDefault()}
+    >
+      <FileBlock padding="m">
+        <Paragraph marginBottomSpacing="xxs">
+          <Text variant="bold" smallScreen>
+            Liitä tai raahaa tiedostot tähän
+          </Text>
+        </Paragraph>
+        <Paragraph marginBottomSpacing="l">
+          <Text smallScreen>Sallitut tiedostomuodot on: xlsx</Text>
```
Comment (kkyttala) [inline src/common/components/new-terminology/info-file.tsx:74]: should this text be localized (also at line 70)

File: src/common/components/new-terminology/missing-info-alert.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  function renderAlert() {
+    if (
+      data.description[0].length === 0 ||
+      !data.mainOrg ||
+      data.infoDomains.length === 0 ||
+      !data.prefix[0] ||
+      data.prefix[1] === false ||
+      !data.contact?.[0] ||
+      data.contact?.[1] === false
+    ) {
```
Comment (kkyttala) [inline src/common/components/new-terminology/missing-info-alert.tsx:43]: contact and description are not mandatory information, but prefLabel should be mandatory

**Review by kkyttala — CHANGES_REQUESTED** [review]

File: src/common/components/new-terminology/language-selector.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+      name: t('languages-name-en'),
+      labelText: t('language-label-text-en'),
+      uniqueItemId: 'en',
+    },
+    {
+      name: t('languages-name-sv'),
+      labelText: t('language-label-text-sv'),
+      uniqueItemId: 'sv',
+    },
+  ];
```
Comment (kkyttala) [inline src/common/components/new-terminology/language-selector.tsx:53]: Application should support other languages too. Ideally we could utilize codelists for this but maybe it's an other feature https://koodistot.suomi.fi/codescheme;registryCode=interoperabilityplatform;schemeCode=languagecodes

File: src/common/components/new-terminology/index.tsx (hunk 1 above)
Comment (topirissanen-atoz) [reply src/common/components/new-terminology/index.tsx:49]: Function for role checking will be added in https://github.com/VRK-YTI/yti-terminology-ui/pull/125

File: src/common/components/new-terminology/language-selector.tsx (hunk 4 above)
Comment (topirissanen-atoz) [reply src/common/components/new-terminology/language-selector.tsx:53]: This is will be implemented later

File: src/common/components/new-terminology/missing-info-alert.tsx (hunk 3 above)
Comment (topirissanen-atoz) [reply src/common/components/new-terminology/missing-info-alert.tsx:41]: Contact has been changed to be optional.
Description variable includes both prefLabel as well as descriptive information about terminology. This checks only that prefLabel is included in the provided information.

File: public/locales/fi/admin.json [hunk 5]
```diff
@@ -77,5 +77,10 @@
   "terminology-description-placeholder": "Kirjoita kuvaus",
   "fi": "suomi",
   "en": "englanti",
-  "sv": "ruotsi"
+  "sv": "ruotsi",
+  "add-or-drag-a-new-file-here": "Liitä tai raahaa tiedosto tähän",
+  "allowed-file-formats": "Sallitut tiedostomuodot on:",
```
Comment (jukkah) [inline public/locales/fi/admin.json:82]: Grammar note: use either "Sallitut tiedostomuodot:" or "Sallitut tiedostomuodot ovat:". The first matches better with English translation.

**Review by jukkah — APPROVED** [review]

**Review by kkyttala — APPROVED** [review]

**Review by jukkah — APPROVED** [review]

**Pull request merged**